File: OilFieldPlatform.Calculation/OilFieldPlatform.Calculation.WebClient/src/api/schemas/requests/WaterSampleEditRequest.ts
export const type = 'pages.waterSample.edit'

export interface WaterSampleEditRequest {
  type: 'pages.waterSample.edit'
  key: string
  property: string
  value: number | null
}

export function create(key: string, property: string, value: number | null): WaterSampleEditRequest {
  return { type, key, property, value }
}

File: OilFieldPlatform.Calculation/OilFieldPlatform.Calculation.Server/Schemas/Requests/WaterSampleEditRequest.cs
using System.Text.Json;

namespace OilFieldPlatform.Calculation.Server.Schemas.Requests;

/// <summary>Запрос редактирования ионного состава пробы.</summary>
public sealed class WaterSampleEditRequest : IWaterSamplePageRequest
{
    /// <summary>Тип сообщения.</summary>
    public string Type { get; init; } = "pages.waterSample.edit";

    /// <summary>Ключ пробы.</summary>
    public required string Key { get; init; }

    /// <summary>Имя редактируемого свойства (Chloride, Carbonate, …).</summary>
    public required string Property { get; init; }

    /// <summary>Новое значение свойства в JSON.</summary>
    public required JsonElement Value { get; init; }
}
